producer/dead_letter_producer.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright © 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package producer contains functions that can be used to produce (that is
+// send) messages to properly configured Kafka broker.
+package producer
+
+import (
+	"github.com/RedHatInsights/insights-results-aggregator/broker"
+	"github.com/Shopify/sarama"
+	"github.com/rs/zerolog/log"
+)
+
+// CreateDeadLetterProducer is a variable for monkey patching in the unit tests
```

Comment:
I've just seen this. Why do you need a `Create` function?

Reply:
The thing is, I need to create mock of DeadLetterProducer  and PayloadTrackerProducer for unit testing. But the variables containing those structs are instantiated in [KafkaConsumer constructor](https://github.com/RedHatInsights/insights-results-aggregator/blob/671708f49047e75bbe0a6a9c7173eecf92ebf983/consumer/consumer.go#L97), so it cannot be easily done e.g. by creating interface.

Reply:
`CreateDeadLetterProducer` is in fact `NewDeadLetterProducer`, but is exchanged for custom function in unit tests.

Reply:
In the end I decided to rename the variable and assign the function directly to the variable without naming it.